regolith/file_system.go

```diff
@@ -872,3 +873,106 @@ func MoveOrCopy(
 	}
 	return nil
 }
+
+// SyncDirectories copies the source to destination while checking size and modification time.
+func SyncDirectories(
+	source string, destination string, makeReadOnly bool, copyParentAcl bool,
+) error {
+	// Make destination parent if not exists
+	destinationParent := filepath.Dir(destination)
+	if err := os.MkdirAll(destinationParent, 0755); err != nil {
+		return burrito.WrapErrorf(err, osMkdirError, destinationParent)
+	}
+	err := filepath.Walk(source, func(srcPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		relPath, err := filepath.Rel(source, srcPath)
+		if err != nil {
+			return burrito.WrapErrorf(err, filepathRelError, source, srcPath)
+		}
+		destPath := filepath.Join(destination, relPath)
+
+		destInfo, err := os.Stat(destPath)
+		if err != nil && !os.IsNotExist(err) {
+			return burrito.WrapErrorf(err, osStatErrorAny, destPath)
+		}
+		if (err != nil && os.IsNotExist(err)) || info.ModTime() != destInfo.ModTime() || info.Size() != destInfo.Size() {
+			if info.IsDir() {
+				return os.MkdirAll(destPath, info.Mode())
+			}
+			Logger.Debugf("SYNC: Copying file %s to %s", srcPath, destPath)
+			return copyFile(srcPath, destPath, info)
+		} else {
+			Logger.Debugf("SYNC: Skipping file %s", srcPath)
+		}
```

Comment:
I think I either don't understand how this whole thing works or I found a problem in the code.

The `SyncDirectories` function runs to sync the file changes from `.regolith/tmp/BP` and `.regolith/tmp/RP` into the export directory. It compares `ModTime()` and `Size()` of each corresponding file. It makes sense when you're syncing two directories where the files are getting modified separately but in Regolith this is not the case. The `tmp` directory is wiped completely every time you run Regolith so the files will have a new `ModTime` even if they are exactly the same as the corresponding files before Regolith run. This means that when syncing the `tmp` with export directory, the entire content of `tmp` is new every time, which means that the `info.ModTime() != destInfo.ModTime()` condition will be always `true` and the file will always be copied.

Reply:
It turns out that the dates are copied from the original files to tmp first.